shm/Player.cpp

```diff
@@ -0,0 +1,61 @@
+#include "Player.hpp"
+#include "Ship.hpp"
+#include "Cargo.hpp"
+#include <memory>
+#include <utility>
+#include <vector>
+#include <iostream>
+
+void Player::calculateAvailableSpace() {
+    const auto& cargoVec = ship_->getCargoVec();
+    if(ship_->getCapacity() <= cargoVec.size()) {
+        availableSpace_ = 0;
+    }
+    availableSpace_ = ship_->getCapacity() - cargoVec.size();
```

Comment:
Remember that this value need to be recalculated each time you add/remove cargo

Reply:
I remember that. Thanks.